src/Interface/HasOrder/Instance.agda

```diff
@@ -12,19 +12,19 @@ import Data.Nat.Properties as NatProp
 
 instance
   preoInt : HasPreorder ℤ _≡_
-  preoInt = record { _≤_ = ℤ._≤_; isPreorder = IntProp.≤-isPreorder }
+  preoInt = hasPreorderFromNonStrict ℤ _≡_ ℤ._≤_ IntProp.≤-isPreorder IntProp._≟_
 
   leqInt : HasPartialOrder ℤ _≡_
-  leqInt = record { hasPreorder = preoInt; antisym = IntProp.≤-antisym }
+  leqInt = record { hasPreorder = preoInt ; ≤-antisym = IntProp.≤-antisym }
 
   DecLeqInt : HasDecPartialOrder ℤ _≡_
   DecLeqInt = record { hasPartialOrder = leqInt ; _≤?_ = ℤ._≤?_ }
 
   preoNat : HasPreorder ℕ _≡_
-  preoNat = record { _≤_ = ℕ._≤_; isPreorder = NatProp.≤-isPreorder }
+  preoNat = hasPreorderFromNonStrict ℕ _≡_ ℕ._≤_ NatProp.≤-isPreorder NatProp._≟_
```

Comment:
The whole point of adding both `_≤_` and `_<_` was that we can supply specialized versions of the two, which may be easier to work with (or faster). This is the prime example where to do it.

Reply:
I don't know what you're suggesting here.  Are you saying I should *not* use `hasPreorderFromNonStrict`?  ...then I don't see the point of having the conversions.

Reply:
I'm saying to use `hasPreorderFromNonStrict` wherever it makes sense, but here the `_<_` of this instance isn't judgementally equal to `Data.Nat._<_`, which I believe is an issue you've mentioned in the past. So the main advantage of this typeclass is that you can make an instance where `_<_` is judgementally equal to `Data.Nat._<_`.

Reply:
Ah! I see.  Than you both for the clarification.  It makes sense now!  🚀